google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1730,6 +1988,168 @@ public TransactionManager transactionManager(TransactionOption... options) {
     }
   }
 
+  class MultiplexedSession implements CachedSession {
+    SessionImpl delegate;
+
+    MultiplexedSession(SessionImpl session) {
+      this.delegate = session;
+    }
+
+    @Override
+    public boolean isAllowReplacing() {
+      // for multiplexed session there is only 1 session, hence there is nothing that we
+      // can replace.
+      return false;
+    }
+
+    @Override
+    public void setAllowReplacing(boolean allowReplacing) {
+      // for multiplexed session there is only 1 session, there is nothing that can be replaced.
+      // hence this is no-op.
+    }
+
+    @Override
+    public void markBusy(ISpan span) {
+      this.delegate.setCurrentSpan(span);
+    }
+
+    @Override
+    public void markUsed() {
+      // no-op for a multiplexed session since we don't track the last-used time
+      // in case of multiplexed session
+    }
+
+    @Override
+    public SpannerException setLastException(SpannerException exception) {
+      // multiplexed sessions run more than one transaction concurrently. we cannot store the
+      // exception state as that is not applicable to all transactions running on the session.
+      return exception;
```

Comment:
Shouldn't this then throw something like an `UnsupportedOperationException`? Or is it OK to call this method for multiplexed sessions, even though we can't store the exception?